app/forms/validators.py

```diff
@@ -438,3 +443,19 @@ def __call__(self, form, field):
 
         if len(data) != 10 or not re.match("^[0-9]+$", data):
             raise validators.ValidationError(self.message)
+
+
+class EmailTldCheck:
+    def __init__(self, message=None):
+        self.message = message or error_messages["INVALID_EMAIL_FORMAT"]
+
+    def __call__(self, form, field):
+        if match := email_regex.match(field.data):
+            hostname = match.group(1)
+            try:
+                hostname = hostname.encode("idna").decode("ascii")
```

Comment:
Discussed with @ajmaddaford and happy this is conforming to validation of the email service provider.